cmd/avrogo/generate.go

```diff
@@ -44,9 +44,14 @@ func generate(w io.Writer, pkg string, ns *parser.Namespace, definitions []schem
 	}
 	// Add avrotypegen package conditionally when there is a RecordDefinition in the namespace.
 	for _, def := range ns.Definitions {
-		if _, ok := def.(*schema.RecordDefinition); ok {
-			gc.addImport("github.com/heetch/avro/avrotypegen")
-			break
+		defToGenerateIdx := sort.Search(len(definitions), func(i int) bool {
```

Comment:
Could you add a test for this?

Reply:
Sure, how should I do that? 
I've extracted that as a function `shouldImportAvroTypeGen`

Reply:
Added some test for you to look at :eye: